src/utils/misc.h

```diff
@@ -30,6 +30,14 @@ struct ExternalImageHashItem {
     int imageWidth;
 };
 
+struct TerminalCmd {
+    QString executablePath;
+    QStringList parameters;
+    QByteArray data;
+    int exitCode;
+    QByteArray resultSet;
+};
+
```

Comment:
I am not too sure about this approach. This object will be emitted in every translation unit `misc.h` gets included in. And `misc` is included in a *lot* of places.

Reply:
Isn't that just a declaration? Should the struct be a typedef?

Reply:
Its both a declaration and a definition. One way to avoid this could be to just forward declare it here and then define it where it is actually being used.

Reply:
OK. Is typedef not used for structs in Qt/C++ anymore?

Reply:
It's not a performance issue at all. Just that it's not being used here, and we could get around it by just defining it in the source file or the place where it actually gets used.

Reply:
I see now what you mean, I'll think about an alternative.